### PR TITLE
Added comments for gateway controller and labels

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

Code snippet that use these constants are very hard to read, IIUC some constants are unnecessary.

### CODE REVIEW

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go [hunk 1]
```diff
@@ -146,16 +142,6 @@ func getClassInfos() map[gateway.GatewayController]classInfo {
...
 	}
 
-	if features.MultiNetworkGatewayAPI {
-		m[constants.UnmanagedGatewayController] = classInfo{
-			// This represents a gateway that our control plane cannot discover directly via the API server.
-			// We shouldn't generate Istio resources for it. We aren't programming this gateway.
-			controller:             constants.UnmanagedGatewayController,
-			description:            "Remote to this cluster. Does not deploy or affect configuration.",
-			disableRouteGeneration: true,
-			addressType:            gateway.HostnameAddressType,
```
Comment (hzxuzhonghu) [inline pilot/pkg/config/kube/gateway/deploymentcontroller.go:156]: cc @stevenctl

File: pkg/config/constants/constants.go [hunk 2]
```diff
@@ -166,11 +166,13 @@ const (
 	WaypointServiceAccount = "istio.io/for-service-account"
 	WaypointForAddressType = "istio.io/waypoint-for"
 
-	ManagedGatewayLabel               = "gateway.istio.io/managed"
-	UnmanagedGatewayController        = "istio.io/unmanaged-gateway"
-	ManagedGatewayControllerLabel     = "istio.io-gateway-controller"
```
Comment (hzxuzhonghu) [inline pkg/config/constants/constants.go:171]: We donot apply this label on kube-gateway,  not sure should we use it later? @howardjohn

**Review by howardjohn — CHANGES_REQUESTED** [review]
I am a bit confused why this is seen as unnecessary?

**Comment by hzxuzhonghu** [comment]
But "istio.io-gateway-controller" is indeed not used now

**Comment by hzxuzhonghu** [comment]
There are so many hacks, how could we invent such a solution? Should try to eliminate

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go [hunk 3]
```diff
@@ -146,16 +142,6 @@ func getClassInfos() map[gateway.GatewayController]classInfo {
 		},
 	}
 
-	if features.MultiNetworkGatewayAPI {
```
Comment (howardjohn) [inline pilot/pkg/config/kube/gateway/deploymentcontroller.go:149]: I am not sure I undertstand. The PR title is "add comments" but we are still deleting real code. Is this meant to be reverted?

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:149]: This seems not needed

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:149]: @stevenctl First added, can you give some insight

**Comment by hzxuzhonghu** [comment]
I am super sure I can not understand the labels/annotations around gateway well, not to mention wide developers except those concentrate on ambient.  I am trying to make it more readable for any beginner to get into ambient feature

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:156]: Yes，https://github.com/istio/istio/blob/c71b83f44e2c82e0c62c3a54f1e21c017cef0a10/pilot/pkg/serviceregistry/kube/controller/network.go#L350
  we get the network address from gateway with `istio-remote` class name. 

But this classinfo is used for convert kube gateway to istio gateway resources. So a gateway with `istio-remote` class name should be skipped here  https://github.com/istio/istio/blob/c71b83f44e2c82e0c62c3a54f1e21c017cef0a10/pilot/pkg/config/kube/gateway/conversion.go#L1986-L1988

**Comment by costinm** [comment]
Do we have a list with all the gateway classes we  implement or support and
their 'stability' and docs / reference about the
API versus implementation details ?

We keep having the implementation details ( with some design docs that may
or may not be accurate ) become a hidden 'API',
along with annotations and env variables.

We were discussing 'swapability' in ambient - running non-Istio Gateways as
waypoints - we need to be very clear with the contracts
and what is expected for conformance and interop. I think the 'remote'
gateway is very important in this context, and not only
for multi-network.

**Comment by hzxuzhonghu** [comment]
@costinm No, at least i cannot find such docs

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go (hunk 3 above)
Comment (stevenctl) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:149]: This is for using Gateway API to specify a remote eastwest gateway without actually deploying things. 

If I have a WorkloadEntry with network: foo and a Gateway with network: foo, traffic to that "workload" goes through the gateway. Allows declaring external destinations without reading an external API server.

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:149]: IIUC, after removing all these stuff, and only keep `istio-remote` does not affect this feature

Comment (stevenctl) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:149]: You're right.. I can't figure out why this was needed.

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:149]: Np, we can remove safely now

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:156]: classInfos is also used for GatewayClass creation

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go (hunk 3 above)
Comment (stevenctl) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:149]: See John's comment. I was under the impression classes were static in the charts but they're generated here.

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:156]: ... why? There is supposed to be a GatewayClass for every class we support. We support istio-remote, so there should be a GWC.

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:156]: GatewayClass is something like ingressclass, which is not a must.

Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:156]: I still dont see why we should remove it. Even if it's not *strictly* required it's still useful?

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:156]: We donot check istio-remote gateway class, nor using it to deploy a gateway.

Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:156]: Yeah GatewayClass is basically documentation. We don't use it in the "istio" class either. 

I'm pretty sure it's required in conformance as well

**Comment by hzxuzhonghu** [comment]
close as can hardly move this forward

**Pull request closed without merging**